packages/react-reconciler/src/ReactProfileTimer.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {Fiber} from './ReactFiber';
+import type {StackCursor, Stack} from './ReactFiberStack';
+
+import warning from 'fbjs/lib/warning';
+
+type Now = () => number;
+
+/**
+ * The "actual" render time is total time required to render the descendants of a Profiler component.
+ * This time is stored as a stack, since Profilers can be nested.
+ * This time is started during the "begin" phase and stopped during the "complete" phase.
+ * It is paused (and accumulated) in the event of an interruption or an aborted render.
+ */
+
+export type ActualRenderTimer = {
+  checkActualRenderTimeStackEmpty(): void,
+  markActualRenderTimeStarted(fiber: Fiber, now: Now): void,
+  pauseActualRenderTimerIfRunning(now: Now): void,
+  recordElapsedActualRenderTime(fiber: Fiber, now: Now): void,
+  resetActualRenderTimer(): void,
+  resumeActualRenderTimerIfPaused(now: Now): void,
+};
+
+export function createActualRenderTimer(stack: Stack): ActualRenderTimer {
```

Comment:
Sorry didn't see that you wrapped this in a later commit. This needs to also include the base time to be safe.

You can pass `now` to this instead of each function.